File: VIQA/HtmlElements/SimpleElements/TextField.cs
using System;
using OpenQA.Selenium;
using VIQA.HtmlElements.Interfaces;

namespace VIQA.HtmlElements
{
    public class TextField : TextElement, ITextField
    {
        public const string LocatorTmpl = "input[type=text][{0}={1}]";
        public static string CommonLocatorById(string id) { return string.Format(LocatorTmpl, "id", id); }
        public static string CommonLocatorByNamed(string id) { return string.Format(LocatorTmpl, "name", id); }
        public static string CommonLocatorByClassName(string id) { return string.Format(LocatorTmpl, "class", id); }

        protected override string _typeName { get { return "TextField"; } }

        public override Func<string> DefaultGetTextFunc { get { return () => GetWebElement().GetAttribute("value"); } }

        public TextField() { }
        public TextField(string name) : base(name) { }
        public TextField(string name, string cssSelector) : base(name, cssSelector) { }
        public TextField(string name, By byLocator) : base(name, byLocator) { }
        public TextField(By byLocator) : base(byLocator) { }
        public TextField(string name, IWebElement webElement) : base(name, webElement) { }
        public TextField(IWebElement webElement) : base(webElement) { }
        
        public void Input(string text)
        {
            DoVIAction("Input text " + text + " in text field", () => GetWebElement().SendKeys(text));
        }

        public void NewInput(string text)
        {
            Clear();
            Input(text);
        }

        public void Clear()
        {
            DoVIAction("Clear text field", () => GetWebElement().Clear());
        }

        public void SetValue<T>(T value)
        {
            if (value == null) return;
            NewInput(value.ToString());
        }

    }
}

File: Java/src/main/java/ru/viqa/ui_testing/elements/simpleElements/TextField.java
package ru.viqa.ui_testing.elements.simpleElements;

import ru.viqa.ui_testing.elements.interfaces.*;
import org.openqa.selenium.*;

import static java.lang.String.format;

/**
 * Created by 12345 on 02.10.2014.
 */
public class TextField extends Text implements ITextField{

    public final static String LocatorTmpl = "input[type=text][%s=%s]";
    public static String commonLocatorById(String id) { return format(LocatorTmpl, "id", id); }
    public static String commonLocatorByNamed(String id) { return format(LocatorTmpl, "name", id); }
    public static String commonLocatorByClassName(String id) { return format(LocatorTmpl, "class", id); }

    public TextField() throws Exception { super(); TypeName = "TextField"; }
    public TextField(String name) throws Exception { super(name); }
    public TextField(String name, String cssSelector) throws Exception  { super(name, cssSelector); }
    public TextField(String name, By byLocator) throws Exception { super(name, byLocator); }
    public TextField(By byLocator) throws Exception { super(byLocator);}
    public TextField(String name, WebElement webElement) throws Exception { super(name, webElement);}
    public TextField(WebElement webElement) throws Exception  { super(webElement);}

    protected void inputAction(String text) throws Exception { getUniqueWebElement().sendKeys(text); }
    protected void clearAction() throws Exception { getUniqueWebElement().clear(); }
    @Override
    protected void setValueAction(String value) throws Exception {
        if (value == null) return;
        newInput(value);
    }

    @Override
    public String getTextAction() throws Exception {
        return getUniqueWebElement().getAttribute("value");
    }

    public final void input(String text) throws Exception {
        doVIAction("Input text '" + text + "' in text area", () -> inputAction(text));
    }

    public final void newInput(String text) throws Exception {
        doVIAction("New input text '" + text + "' in text area", () -> {
            clearAction();
            inputAction(text);
        });
    }

    public final void clear() throws Exception {
        doVIAction("Clear text area", this::clearAction);
    }
}
